### PR TITLE
Add error message for counterfactual panel

### PR DESCRIPTION
**Pull request by tongyu-microsoft**
When user hits errrors in counterfactual panel, we should be able to display error messages in the counterfactual panel.
image

## Description
This PR is to add error message in counterfactual panel. So when user hits /predict call errors, we can display error messages in counterfactual panel in AML studio.

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [x] responsibleai/core-ui
- [x] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] erroranalysis
- [ ] raiutils
- [ ] raiwidgets
- [ ] rai_core_flask
- [ ] responsibleai

## Tests

- [ ] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Screenshots (if appropriate):
image

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by imatiach-msft — APPROVED** [review]

File: libs/core-ui/src/lib/Interfaces/ICounterfactualData.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 export interface ICounterfactualData {
   cfs_list: Array<Array<Array<string | number>>>;
+  error_message?: string;
```
[the remark this answers is not in the document]
Comment (tongyu-microsoft) [reply libs/core-ui/src/lib/Interfaces/ICounterfactualData.ts:6]: Just updated it in camel case. Interesting that only the new added properties require camel case? I was trying to follow the style of other properties :)

**Comment by tongyu-microsoft** [comment]
> What is the error message here? Seems like from the screenshot that CF were generated then why the error "No counterfactuals were generated" displayed?

Hi Gaurav, there is no issue with CF. Here we want to add capability to render error messages. I put a error message placeholder here to for screenshot to display how it looks like when error message is rendered. We will pass the error message string from AML, and display it in CF panel when we hit /predict call errors in AML

**Comment by gaugup** [comment]
> > What is the error message here? Seems like from the screenshot that CF were generated then why the error "No counterfactuals were generated" displayed?
> 
> Hi Gaurav, there is no issue with CF. Here we want to add capability to render error messages. I put a error message placeholder here to for screenshot to display how it looks like when error message is rendered. We will pass the error message string from AML, and display it in CF panel when we hit /predict call errors in AML

Thanks for the clarification.

**Review by gaugup — APPROVED** [review]

File: libs/core-ui/src/lib/Interfaces/ICounterfactualData.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (imatiach-msft) [reply libs/core-ui/src/lib/Interfaces/ICounterfactualData.ts]: oh, there are some values whitelisted somewhere:
https://github.com/microsoft/responsible-ai-toolbox/blob/ce9521aea4b1ce32b6337a1aed5f4596397f2ca1/.eslintrc/.eslintrc.custom.eslintrc#L44

if you want camelcase you can add it to the whitelist

**Pull request merged**